### PR TITLE
Change network handling to example42/network

### PR DESCRIPTION
**Pull request by cgoubert**
We use https://forge.puppet.com/example42/network to control bridge setup in a more flexible way.

### CODE REVIEW

File: manifests/install.pp [hunk 1]
```diff
@@ -15,14 +15,27 @@
...
-    content => template('proxmox/interfaces.erb'),
-    owner   => 'root',
-    group   => 'root',
-    mode    => '0644'
+  network::interface { 'vmbr0':
+    family       => 'inet',
+    ipaddress    => $::ipaddress,
+    netmask      => $::netmask,
+    gateway      => $gatewayv4,
+    bridge_ports => [ $::netdev ],
```
Comment (cornuwel) [inline manifests/install.pp:23]: This breaks CI:
manifests/install.pp - WARNING: top-scope variable being used without an explicit namespace on line 22

File: manifests/install.pp [hunk 2]
```diff
@@ -15,14 +15,27 @@
...
+    bridge_stp   => 'off',
+    bridge_fd    => 0,
   }
-  reboot { '/etc/network/interfaces':
-    subscribe => [ File['/etc/network/interfaces'], Package['proxmox-ve'] ],
```
Comment (cornuwel) [inline manifests/install.pp:26]: Maybe just subscribe to the proxmox-ve package. The ordering shoud make sure it happens after the network config.

**Review by cornuwel — COMMENTED** [review]
You can also delete the erb template from the module.
Note for your tests, you can just edit the puppetfile in 'puppet' and tell it to use your github branch of puppet-proxmox. Then run r10k.

File: manifests/install.pp (hunk 2 above)
Comment (cgoubert) [reply manifests/install.pp:26]: Resolved by https://github.com/SineQuaNonSoftware/puppet-proxmox/pull/7/commits/67658a5faa077d46e715658a86407109aad1d146

File: manifests/install.pp [hunk 3]
```diff
@@ -11,18 +11,38 @@
     ensure => 'absent'
   }
 ->package { ['proxmox-ve', 'postfix', 'open-iscsi']:
-    ensure => 'installed'
+    ensure => 'installed',
+    notify => Reboot['proxmox_install'],
```
Comment (cornuwel) [inline manifests/install.pp:15]: Didn't know that syntax worked

File: manifests/install.pp [hunk 4]
```diff
@@ -11,18 +11,38 @@
...
+      'echo 1 > /proc/sys/net/ipv4/ip_forward',
+      'iptables -t nat -A POSTROUTING -s \'10.0.1.0/24\' -o vmbr0 -j MASQUERADE',
+    ],
+    post_down    => [
+      'iptables -t nat -D POSTROUTING -s \'10.0.1.0/24\' -o vmbr0 -j MASQUERADE',
+    ],
+  }
+
+  reboot { 'proxmox_install':
+    apply =>  finished, # Wait until entire catalog is applied before rebooting
```
Comment (cornuwel) [inline manifests/install.pp:46]: Much better 😃

**Review by cornuwel — APPROVED** [review]
Looks fine to me, do you want do a full test by pointing the puppetmaster to your branch and doing a full reinstall?

**Comment by cgoubert** [comment]
> Looks fine to me, do you want do a full test by pointing the puppetmaster to your branch and doing a full reinstall?

I don't think so, I'm still working on terraform deployments on the server at the moment.

**Comment by cornuwel** [comment]
Well, I'll say let's merge it and deal with any eventual issue when we deploy proxmox1.sinequanon.io, in 2 years 💯

**Pull request merged**